### ISSUE TITLE
about how to compute pairwise_potential_E

### ISSUE THREAD

**Issue by BeHappy1** [baidu-research/NCRF#13]
hi,may i ask about why you compute "pairwise_potential_E" by minus
pairwise_potential_E = torch.sum(
               **probs** * **pairwise_potential - (1 - probs)** * **pairwise_potential**,
                dim=2, **keepdim=True)
if i want to use it for multi-classification,how could i modify it.
thanks.

**Comment by yil8** [comment]
@BeHappy1 since quite a few people were asking the details of CRF module implementation, I've updated the doc string of [wsi/model/layers.py](https://github.com/baidu-research/NCRF/blob/master/wsi/model/layers.py#L19). Since we were doing binary classification, 1-d logit was used, where > 0 means tumor and < 0 means normal, and binary cross entropy loss was used. If you want to go multi-class, I guess you have to use N-d logits, where N is the number of classes, and softmax has to be used for the loss. I haven't thought about the details yet, but I think if should be minor modification on top of the current implementation, mostly on the expectation and normalization part.

**Comment by BeHappy1** [comment]
thanks so much,i got it.
